modules/wordads/php/widgets.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Widget for inserting an ad into your sidebar
+ *
+ * @since 0.1
+ * TODO: sanitize all the output
+ */
+class WordAds_Sidebar_Widget extends WP_Widget {
+
+	private static $allowed_tags = array( 'mrec', 'wideskyscraper' );
+
+	private $options = array();
+
+	function __construct() {
+		parent::__construct(
+			'wordads_sidebar_widget',
+			__( 'WordAds Widget (Jetpack)', 'jetpack' ),
+			array( 'description' => __( 'Insert a WordAds ad wherever you can place a widget.', 'jetpack' ) )
```

Comment:
@dbspringer there are roughly 3 things to keep in mind to add selective refresh:
1. add the `'customize_selective_refresh' => true`

2. make sure no styles or scripts are enqueued in the `->widget()` method. A method should be hooked to `wp_enqueue_scripts` in the `__constructor`, enqueing it when `is_customize_preview()`

3. if there's JS, it should be re-executed when the widget is refreshed. This is the tricky part and varies from JS to JS based on what is needed.
See this commit for an example where I added it to the new Google Translate widget
https://github.com/Automattic/jetpack/pull/5440/commits/b58b1934a98e7b367d0b496847cc9bbb674d80f6
since the code was run when the script from Google was loaded, I had to remove the script tag and add it again so it worked.
Others don't need such an esoteric solution, like
https://github.com/Automattic/jetpack/pull/3607/files#diff-4f9b83c204105157a58e1d538ab4925f
where the gallery is rebuilt when the event `'partial-content-rendered'` is triggered. See this last PR for some examples, this is when SR support was added to all widgets.
Let me know if you need a hand, will be glad to help you.